osmorc/src/org/osmorc/facet/OsmorcFacetConfiguration.java

```diff
@@ -278,6 +283,13 @@ public void writeExternal(Element element) throws WriteExternalException {
       additionalJARContentsElement.addContent(entry);
     }
     element.addContent(additionalJARContentsElement);
+
+    // reset the background build process to pick up facet configuration changes
+    if (myModificationCount.get() > 0) {
+      BuildManager.getInstance().clearState(myFacet.getModule().getProject());
+    }
+
+    myModificationCount.set(0);
```

Comment:
The value is used for cache invalidation and is expected to increase monotonically; resets may cause hard-to-reproduce problems.